packages/tracing/src/browser/metrics.ts

```diff
@@ -0,0 +1,279 @@
+    if (!global || !global.performance || !global.performance.getEntries) {
+      // Gatekeeper if performance API not available
+      return;
+    }
+
+    logger.log('[Tracing] Adding & adjusting spans using Performance API');
+
+    // TODO(fixme): depending on the 'op' directly is brittle.
+    if (transaction.op === 'pageload') {
+      // Force any pending records to be dispatched.
+      this._forceLCP();
+      if (this._lcp) {
+        // Set the last observed LCP score.
+        transaction.setData('_sentry_web_vitals', { LCP: this._lcp });
+      }
+    }
+
+    const timeOrigin = msToSec(performance.timeOrigin);
+    let entryScriptSrc: string | undefined;
+
+    if (global.document) {
+      // tslint:disable-next-line: prefer-for-of
+      for (let i = 0; i < document.scripts.length; i++) {
+        // We go through all scripts on the page and look for 'data-entry'
+        // We remember the name and measure the time between this script finished loading and
+        // our mark 'sentry-tracing-init'
+        if (document.scripts[i].dataset.entry === 'true') {
+          entryScriptSrc = document.scripts[i].src;
+          break;
+        }
+      }
+    }
+
+    let entryScriptStartEndTime: number | undefined;
+    let tracingInitMarkStartTime: number | undefined;
+
+    global.performance
+      .getEntries()
+      .slice(this._performanceCursor)
+      .forEach((entry: Record<string, any>) => {
+        const startTime = msToSec(entry.startTime as number);
+        const duration = msToSec(entry.duration as number);
+
+        if (transaction.op === 'navigation' && timeOrigin + startTime < transaction.startTimestamp) {
+          return;
+        }
+
+        switch (entry.entryType) {
+          case 'navigation':
+            addNavigationSpans(transaction, entry, timeOrigin);
+            break;
+          case 'mark':
+          case 'paint':
+          case 'measure':
+            const startTimestamp = addMeasureSpans(transaction, entry, startTime, duration, timeOrigin);
+            if (tracingInitMarkStartTime === undefined && entry.name === 'sentry-tracing-init') {
+              tracingInitMarkStartTime = startTimestamp;
+            }
+            break;
+          case 'resource':
+            const resourceName = (entry.name as string).replace(window.location.origin, '');
+            const endTimestamp = addResourceSpans(transaction, entry, resourceName, startTime, duration, timeOrigin);
+            // We remember the entry script end time to calculate the difference to the first init mark
+            if (entryScriptStartEndTime === undefined && (entryScriptSrc || '').indexOf(resourceName) > -1) {
+              entryScriptStartEndTime = endTimestamp;
+            }
+            break;
+          default:
+          // Ignore other entry types.
+        }
+      });
+    if (entryScriptStartEndTime !== undefined && tracingInitMarkStartTime !== undefined) {
```

Comment:
`startEndTime` is very confusing name tbh

Reply:
Good point, but I don't really have anything better for a name. Maybe @HazAT can help us here?

Reply:
I think we can just call it `entryScriptStartTimestamp`.
On line 157 we set `endTimestamp` of the script but it's fine (since it's the time we finished downloading the script).